velox/expression/CastExpr-inl.h

```diff
@@ -301,6 +301,68 @@ VectorPtr CastExpr::applyDecimalToBooleanCast(
   return result;
 }
 
+template <typename FromNativeType>
+VectorPtr CastExpr::applyDecimalToVarcharCast(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& fromType) {
+  VectorPtr result;
+  context.ensureWritable(rows, VARCHAR(), result);
+  (*result).clearNulls(rows);
+  const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
+  auto [precision, scale] = getDecimalPrecisionScale(*fromType);
+  // Calculate the max total size of the raw string buffer.
+  size_t maxTotalResultBytes = rows.countSelected() * precision;
```

Comment:
> A decimal number has a precision and scale, hence, we know the max size of the serialized string and therefore can pre-allocate sufficient buffer.

@mbasmanova The allocated buffer size could be larger than the actual needed size, and do we need to calculate the precise size based on the input value through e.g. log10 as https://github.com/facebookincubator/velox/pull/6210#discussion_r1303687705?

Reply:
It can be larger indeed, but we will know the exact size after writing all the rows and can update if needed. Although it is not necessary to specify the exact size anyway. It can be larger, just cannot be shorter.

Reply:
Changed to allocate as max size and update the exact size finally.